osi_hostvehicledata.proto

```diff
@@ -2,31 +2,369 @@ syntax = "proto2";
+    }
+
+    //
+    // \brief The focus here is on the description of the wheels.
+    //
+    message VehicleWheels
+    {
+        // Contains the description of a wheel.
+        //
+        repeated Wheel wheel = 1;
+
+        //
+        // \brief The focus here is on the description of a wheel.
+        //
+        message Wheel
+        {
+            // The axle which contains this wheel.  A value of 0 represents the
+            // front-most axle of the vehicle with higher numbers incrementing
+            // towards the rear-most axle.
+            //
+            optional uint32 axle = 1;
+
+            // The index of the wheel on the axle, counting in the direction
+            // of positive-y, i.e. right-to-left.
+            //
+            // For example, on a standard 2-axle, 4-wheel car, the rear-right
+            // wheel would be (axle=1, index=0).
+            //
+            optional uint32 index = 2;
+
+            // Dry friction is a force that opposes the relative lateral motion of two solid surfaces 
+            // in contact. It is subdivided into static friction between non-moving surfaces and kinetic
+            // friction between moving surfaces.
+            // Ued here is the dry friction coefficient of the paired materials (see reference).
+            // Dimensionless.
+            //
+            // Unit: -
+            //
+            // \par References:
+            // - http://adaptivemap.ma.psu.edu/websites/6_friction/dry_friction/dryfriction.html
+            //
+            //optional double kinetic_friction_coefficient = 3;
+
+            // Contact point of the mid of the tire tread with the world.
+            // Absolute coordinates (x, y, z).
+            //
+            // Unit: m
+            //
+            //optional Vector3d contact_point = 4;
+
+            // Contains the rotation rate of the wheel per second in radians.
+            // Based on the processed output of the hall sensor at the wheel.
+            //
+            // Unit: rad/s
+            //
+            optional double rotation_rate = 5;
+
+            // Contains the longitudinal slip of the tire.
+            // \par References:
+            // - https://www.kfz-tech.de/Biblio/Formelsammlung/Schlupf.htm
+            //
+            // Unit: %
+            //
+            optional double slip = 6;
+        }
+    }
+
+    //
+    // \brief This message contains all the information the vehicle knows about its kinematic states.
+    //
+    message VehiclePositionAndKinematics
+    {
+        //
+        // \brief Current kinematic data based on the output of the Intertial Measurement Unit (IMU).
+        //
+        message CartesianInformation
+        {
+            // Current kinematic data based on the output of the Intertial Measurement Unit (IMU).
+            //
+            // \note Note that dimension and base_polygon need not be set.
+            //
+            optional BaseMoving cartesian_data = 1;
```

Comment:
We have GeoreferencedInformation which reads it is the output of the GPS Unit. Now my question is: If I have a localization algorithm that estimates my global position. Then were do I put this information? GeoreferencedInformation is reserved for GNSS and CartesianInformation is reserved for IMU. 

Also let's assume the algorithm provides the position in ENU according to ISO8855. This means I have a cartesian  coordinate system but I also need to provide the coordinates of the coordinate system origin so that I am able to convert between ENU and geodetic position. Even if it is allowed to write this information to either IMU or GNSS. In GeoreferencedInformation I do not have a 3d orientation (at most commented out yaw). And in CartesianInformation I am missing the geodetic coordinates of the origin.

Reply:
@u21n80 can you add your suggestions as a pull request onto this branch?
@ThomasNaderBMW would that be ok for you to add pull request onto this branch?

Reply:
Now I made it more generic.

@u21n80 : 
1. Regarding ENU: Does it help to add this as reference?
2. And also to add the startpoint x,y,z? What is the startpoint, the engine start oder simulation start
3. Do we need x,y,z travelled?
3. Why do you need a yaw value? In geodetic description?
==> Many questions, a topic to discuss @ThomasSchloemicherAVL

Reply:
@ThomasNaderBMW 
1. adding the ENU reference helps a lot. In simulation this might not be required as simulation often times (not always) uses an arbitrary cartesian coordinate system, but without the origin you are loose the information about how to convert between the cartesian (local) coordinate frame and a geodetic position. E.g. you could not compare your data against other geodetic data, e.g. reference Gnss measurements or even overlays with a map are not possible as well without knowing the origin of your coordinate system. 
2. Having a startpoint in cartesian coordinate is not equivalent to having the ENU origin see point 1. I do not know what startpoint you are refering to. If this is in regard to a reference point: It can be chosen anyway you like, but depending on what you want to achieve both could make sense. From the viewpoint of simulation choosing the origin at the start of the simulation might make sense. If you want to compare odometry estimates (which starts when the vehicle starts), then vehicle start might make more sense. The important part is that a  reference is specified, not necessarly where.
3. xyz travelled is not very meaningful I think, as when I drive forward and then backward again, my xyz travelled distance would  be zero. We could use the distance travelled instead, but for me we define a pose of the vehicle and a pose does not have a distance associated to it. If you were to store the previous poses as well (basically the path the vehicle drove) then I would agree that the distance is a property of the path. Maybe as a compromise we could a uptime field which stores the duration how long the vehicle was active and how far it drove kind of like a diagnostic field. But I do not see it as part of the pose.
4. In my pull request #488 I already removed the geodetic information (including the yaw angle). But this pulll request is not yet approved.
Hope this feedback is useful to you.